[waihongteh] iP

waihongteh opened the pull request:
# Waterfall
> “Your mind is for having ideas, not holding them.” – [David Allen](https://dansilvestre.com/productivity-quotes/)

Waterfall frees your mind of having to remember things you need to do. It's,

- **text-based**
- _easy to learn_
- ~~FAST~~ ***SUPER FAST*** to use

All you need to do is:

1. [Download it from here](https://github.com/waihongteh/ip/releases/tag/A-JAR).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉.

And it is **FREE**!

### Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

waihongteh committed:
Change chatBot name and greeting message

waihongteh committed:
Change file and class name to chatBot name

waihongteh committed:
Separate welcome and bye message

waihongteh committed:
Implement scanner and echo system

waihongteh committed:
Add indentation

waihongteh committed:
Add personality to the bot

waihongteh committed:
Add list functionalities

waihongteh committed:
Switch to switch method instead of conditional statement for better readability

waihongteh committed:
Add Task class implementation

waihongteh committed:
Edit Task class implementation and finish chatbot mark and unmark functionalities

waihongteh committed:
Add todo class

waihongteh committed:
Add deadline class

waihongteh committed:
Add Event class

waihongteh committed:
Finish integration with todo, deadline and event for chatbot

waihongteh committed:
Finish A-TextUiTesting

waihongteh committed:
Add waterfall exception class

waihongteh committed:
Add exception handling for possible errors in text inputs

waihongteh committed:
Add delete functionalities

waihongteh committed:
Solve bug where deadline without a /by command leads to error

waihongteh committed:
Add local data storage and file manager class

waihongteh committed:
Update gitignore for data files

waihongteh committed:
Merge branch 'branch-Level-7'

waihongteh committed:
Add date time parser for storage

waihongteh committed:
Merge branch 'branch-Level-8'

waihongteh committed:
Add OOP classes

waihongteh committed:
Organise into packages

waihongteh committed:
Merge branch 'add-gradle-support' of https://github.com/waihongteh/ip into add-gradle-support

waihongteh committed:
Merge branch 'add-gradle-support'

waihongteh committed:
Integrate gradle

waihongteh committed:
Add junit tests

waihongteh committed:
Modify gradle build option for JAR

waihongteh committed:
Add Javadoc to all classes and methods

waihongteh committed:
Merge branch 'branch-A-JavaDoc'

waihongteh committed:
Modify code based on coding standard

waihongteh committed:
Merge branch 'branch-A-CodingStandard'

waihongteh committed:
Add find functionalities

waihongteh committed:
Add find functionalities

waihongteh committed:
Merge branch 'branch-Level-9'

waihongteh committed:
Change readme name

waihongteh committed:
change readme name

waihongteh committed:
Merge branch 'branch-Level-9'

waihongteh committed:
Add checkstyle

waihongteh committed:
Add checkstyle

waihongteh committed:
Merge branch 'master' of https://github.com/waihongteh/ip

charlesliyifeng commented on src/main/java/waterfall/Parser.java:
```diff
@@ -0,0 +1,122 @@
+ *
+ * @author Wai Hong
+ */
+
```
Should there be a blank line between the documentation block and the class?

charlesliyifeng commented on src/main/java/waterfall/Storage.java:
```diff
@@ -0,0 +1,114 @@
+                }
+                String[] strings = line.split(" \\| ");
+                Task task;
+                switch (strings[0]) {
```
Good job on following the correct indentation for switch statements!

charlesliyifeng commented on src/main/java/waterfall/Ui.java:
```diff
@@ -0,0 +1,129 @@
+    }
+
+    /**
+     * Display an error message to the user.
```
Should this be "Displays" instead of "Display"?

charlesliyifeng commented on src/main/java/waterfall/command/MarkCommand.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws WaterfallException, IOException {
+        if (!tasks.checkIndex(index)) {
+            throw new WaterfallException("Why are you trying to edit a waterfall waterfall.task that does not exist?");
```
This line might be a little too long?

charlesliyifeng commented on src/main/java/waterfall/task/Event.java:
```diff
@@ -0,0 +1,76 @@
+    @Override
+    public String toStorageString() {
+        if (this.isDone()) {
+            return "E | 1 | " + this.getTitle() + " | " + getFormattedTime(start) + " | " + getFormattedTime(end);
```
Maybe add some line breaks to improve readability?

charlesliyifeng commented on src/test/java/waterfall/ParserTest.java:
```diff
@@ -0,0 +1,70 @@
+package waterfall;
+
+import org.junit.jupiter.api.Test;
+import waterfall.command.*;
```
Should the imports be listed explicitly?

charlesliyifeng reviewed (commented):
LGTM, just some minor issues with coding standard.

waihongteh committed:
Add GUI functionality

waihongteh committed:
Add better styling to GUI

waihongteh committed:
Merge branch 'branch-Level-10'

zhiyi12345 commented on src/main/java/waterfall/Parser.java:
```diff
@@ -0,0 +1,122 @@
+     * @return The corresponding <code>Command</code> object based on the user input.
+     * @throws WaterfallException if the input format is invalid or if required fields are missing.
+     */
+    public static Command parse(String input) throws WaterfallException {
```
Perhaps you could work on the reducing the number of if-else statements to improve readability?

zhiyi12345 commented on src/test/java/waterfall/StorageTest.java:
```diff
@@ -0,0 +1,78 @@
+package waterfall;
+
+import org.junit.jupiter.api.*;
```
Perhaps you could specify the specific class to import?

zhiyi12345 commented on src/test/java/waterfall/StorageTest.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Test
+    void load_shouldLoadTasksFromFile() throws IOException, WaterfallException {
```
Perhaps you should use the CamelCase format here too!

zhiyi12345 reviewed (commented):
Just some nits to fix. I like how your code is properly java documented!

waihongteh committed:
Add stream method for list

waihongteh committed:
Add assertions

There are several methods that would crash if unexpected behaviour occurs to some constants that are non-null.

This commit adds a null check by adding Java assertion before these possible positions.

waihongteh committed:
Improve code quality

This commit checks the code quality and apply changes if necessary to comply with the coding standard.

Some methods have been refactored for better readability.

waihongteh committed:
Merge pull request #1 from waihongteh/branch-A-Assertions

Add assertions

waihongteh committed:
Solve merge conflict with master

waihongteh committed:
Merge branch 'master' into branch-A-Streams

waihongteh committed:
Merge pull request #2 from waihongteh/branch-A-CodeQuality

Improve code quality

waihongteh committed:
Merge branch 'master' into branch-A-Streams

waihongteh committed:
Merge pull request #3 from waihongteh/branch-A-Streams

Add stream method for list

waihongteh committed:
Add undo functionality

Undo functionality is now added to cater better users' experience.

This commit adds command "undo" to undo previous commands that are executed.

waihongteh committed:
Merge branch 'master' of https://github.com/waihongteh/ip

waihongteh committed:
Refactor parser method

waihongteh committed:
Update ui test cases

waihongteh committed:
Improve GUI

waihongteh committed:
Add Ui image and edit user guide
